### PR TITLE
feat: update table output

### PR DESCRIPTION
**Pull request by rossiam**
Update table output:
	- switch to table package which handles international characters properly
	- removed compact / expanded command line options
	- removed compactTableOutput configuration option
	- added group-rows and no-group-rows command line options
	- added groupTableOutputRows configuration option
	- (lib) completely isolated use of dependency to table-generator.ts

## Checklist

- [x] I have read the **[CONTRIBUTING](../CONTRIBUTING.md)** document
- [x] Any required documentation has been added
- [x] My code follows the code style of this project (`npm run lint` produces no warnings/errors)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: packages/lib/src/__tests__/table-generator.test.ts [hunk 1]
```diff
@@ -49,50 +49,45 @@ function stripEscapeSequences(str: string): string {
...
-function rowCount(table: string): number {
-	return table.split('\n').length - 2
-}
+/**
+ * Clean up string created using backticks. Lines are stripped of ` +|` at the beginning of each
+ * line (allowing indentation) and `|` at the end (allowing for spaces at the end of the line
+ * that are not automatically removed by tooling). If there is a new-line at the beginning of the
+ * string, it is also removed.
+ */
+const fixIndent = (input: string): string => input.replace(/^\s*\|/gm, '').replace(/\|$/gm, '').replace(/^\n/, '')
```
Comment (john-u) [inline packages/lib/src/__tests__/table-generator.test.ts:58]: It is not clear to me why I would need this if I was adding tests here. Is there something preventing us from formatting the expected strings without a function?

File: packages/cli/package.json [hunk 2]
```diff
@@ -75,14 +75,13 @@
 		"@smartthings/core-sdk": "^4.2.0",
 		"@smartthings/plugin-cli-edge": "^1.14.1",
 		"aws-sdk": "^2.1144.0",
-		"cli-table": "^0.3.11",
 		"inquirer": "^8.2.4",
 		"js-yaml": "^4.1.0",
-		"log4js": "6.3.0"
+		"log4js": "6.3.0",
+		"table": "^6.8.0"
```
Comment (john-u) [inline packages/cli/package.json:81]: It doesn't look like this is actually required for the cli package.

**Review by john-u — COMMENTED** [review]
Awesome changes!

File: packages/cli/package.json (hunk 2 above)
Comment (rossiam) [reply packages/cli/package.json]: Good catch. Fixed.

File: packages/lib/src/__tests__/table-generator.test.ts (hunk 1 above)
Comment (rossiam) [reply packages/lib/src/__tests__/table-generator.test.ts:58]: It just makes the tests look nicer. We can use the mutli-line backtick quoting and still indent and not worry about the editor stripping the whitespace from the end of the line.

**Review by john-u — APPROVED** [review]

**Pull request merged**